### PR TITLE
[T4A4][T15-A2]Amro Shohoud

### PR DESCRIPTION
**Pull request by AmroShohoud**

### CODE REVIEW

**Comment by AmroShohoud** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -35,6 +35,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
+    
+    @Override
+    public String getPrintableString() {
+        return "Address: " + toString();
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Address.java:41]: Magic string? `"Address: "`

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -78,6 +78,17 @@ public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
         return Objects.hash(name, phone, email, address, tags);
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    public String getPrintableString(Printable... printables) {
+       String printableString = "";
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Person.java:86]: Can consider use `StringBuffer` here.

File: src/seedu/addressbook/data/person/Person.java [hunk 3]
```diff
@@ -78,6 +78,17 @@ public int hashCode() {
...
+       for (Printable printable : printables) {
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Person.java:87]: Good usage of `foreach` loop!

File: src/seedu/addressbook/data/person/Printable.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package seedu.addressbook.data.person;
+
+/**
+ * An interface for printing Person data
+ */
+
```
Comment (xpdavid) [inline src/seedu/addressbook/data/person/Printable.java:6]: The blank line is not necessary.

**Comment by xpdavid** [comment]
@AmroShohoud Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**